### ISSUE TITLE
whitespace in message custom_properties raises MQTT ValueError

### ISSUE THREAD

**Issue by meshell** [Azure/azure-iot-sdk-python#503]
```
Unexpected error in <azure.iot.device.common.pipeline.pipeline_stages_mqtt.MQTTTransportStage object at 0x7f51ebf0b710>._run_op() call
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/azure/iot/device/common/pipeline/pipeline_stages_base.py", line 102, in run_op
    self._run_op(op)
  File "/usr/local/lib/python3.7/site-packages/azure/iot/device/common/pipeline/pipeline_thread.py", line 198, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 159, in _run_op
    self.transport.publish(topic=op.topic, payload=op.payload, callback=on_published)
  File "/usr/local/lib/python3.7/site-packages/azure/iot/device/common/mqtt_transport.py", line 533, in publish
    (rc, mid) = self._mqtt_client.publish(topic=topic, payload=payload, qos=qos)
  File "/usr/local/lib/python3.7/site-packages/paho/mqtt/client.py", line 1245, in publish
    raise ValueError('Publish topic cannot contain wildcards.')
ValueError: Publish topic cannot contain wildcards.

MQTTPublishOperation: completing with error Publish topic cannot contain wildcards.
SendOutputEventOperation: completing with error Publish topic cannot contain wildcards.
Callback completed with error Publish topic cannot contain wildcards.
['ValueError: Publish topic cannot contain wildcards.\n']
```

**Comment by jebrando** [comment]
@meshell Can you give some a code snippet showing what you were trying to do.  Thank you.

**Comment by meshell** [comment]
Simply do something like this:
`new_message = Message("Hello world")
new_message.custom_properties['topic'] = "major topic | minor topic"`

**Comment by jebrando** [comment]
@meshell We are working on a fix for this issue and should be ready soon.

**Comment by cartertinney** [comment]
This issue has been fixed as of azure-iot-device version 2.1.1

**Comment by az-iot-builder-01** [comment]
@meshell, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)

**Comment by amoghstayade** [comment]
Faced the same issue with azure-iot-device 2.0.1. Fixed it with 2.1.1. Thank you for the contribution.
